Automattic/jetpack
# Social: Add library for Social Image Generator

## Proposed changes:

This PR introduces a library to handle all Social Image Generator-related functionality on the Jetpack side. In this initial iteration, it consists of two separate class: `Setup`, which sets up the library and registers hooks, and `Extractor`, a standalone class that can be used to extract SIG-related information from a post.

Currently, all the library does is retrieving a token from WPCOM on post save and saving this token to the post meta.

### Other information:


## Jetpack product discussion


1202816441505085-as-1203975822898236/f

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






~~_Note: this can't really be tested until https://github.com/Automattic/jetpack/pull/28737 is merged_~~

* ~~Sandbox D101988-code and point your site to it~~
* Create a new post and enable Social Image Generator.
* Save the post.
* Check the post meta and make sure the token is added to the `_wpas_options` meta key.
  * `jetpack docker wp post meta list POST_ID`
* Make sure all the tests pass.
  *  `cd projects/packages/publicize && composer run test-php`
